### PR TITLE
Issue483

### LINKED ISSUE #482: Metagenomics Sample Sheet generation introduces spaces in 'Sample_project' column.
Lab Control introduces space characters in 'Sample_Project' column because it appends the Qiita study owner. Spaces are not permitted within this column for the downstream workflows (this field specifies the directory where the demultiplexed data will be deposited). 

Screen Shot 2019-04-24 at 10 56 57 AM

### PR DESCRIPTION
**Pull request by AmandaBirmingham**
Fixes #482 (see new `_folder_scrub_name` and changes to `_generate_sample_proj_value`, and new `test__folder_scrub_name`) and #483 (see changes to `_generate_shotgun_sample_sheet` and `_generate_sample_proj_value`, and revised `test_generate_sample_sheet_shotgun`).  

Also preps `test_process.py` for forthcoming work on #438 (see new `pooling-process-echo-picklist.txt` file and revised `test_generate_echo_picklist_default`),

### CODE REVIEW

File: labcontrol/db/tests/test_process.py [hunk 1]
```diff
@@ -1493,6 +1490,12 @@ def test_bcl_scrub_name(self):
         self.assertEqual(SequencingProcess._bcl_scrub_name('test-1'), 'test-1')
         self.assertEqual(SequencingProcess._bcl_scrub_name('test_1'), 'test_1')
 
+    def test__folder_scrub_name(self):
+        input= "Allison  Vrbanc-Meade*,_Pat O'Brien_1"
```
Comment (charles-cowart) [inline labcontrol/db/tests/test_process.py]: Is it okay to use Allison's name here?

Comment (AmandaBirmingham) [reply labcontrol/db/tests/test_process.py]: Good point!  I'll change that.

**Comment by AmandaBirmingham** [comment]
Ping @wasade Do you know when you will have availability to code review this PR? We need it for today's sprint, and it is also the basis of a forthcoming PR that we also need for this sprint for issue #438 + #505 .

File: labcontrol/db/tests/test_process.py [hunk 2]
```diff
@@ -1493,6 +1490,12 @@ def test_bcl_scrub_name(self):
         self.assertEqual(SequencingProcess._bcl_scrub_name('test-1'), 'test-1')
         self.assertEqual(SequencingProcess._bcl_scrub_name('test_1'), 'test_1')
 
+    def test__folder_scrub_name(self):
+        input= "Ogden  Bogden-Meade*,_Pat O'Brien_1"
```
Comment (wasade) [inline labcontrol/db/tests/test_process.py]: Minor, but I think flake8 may twitch here with the missing space prior to the `=`

**Comment by wasade** [comment]
@charles-cowart, are you :+1:?

**Comment by charles-cowart** [comment]
Good, thanks! My apologies for the wait.

File: labcontrol/db/tests/test_process.py (hunk 2 above)
Comment (AmandaBirmingham) [reply labcontrol/db/tests/test_process.py]: Thanks!  If you don't mind, I'll take care of this in the next PR ... :D

**Pull request merged**